### PR TITLE
🪟🎨 Connector form: Improve logs look and feel

### LINKED ISSUE #20238: Connector form: Improve logs rendering on failing connection test
With advanced mode enabled, the logs from the connection test job are shown below the form together with the error message:
Screenshot 2022-12-08 at 17 19 15

This UI is confusing users because it looks like it failed and worked at the same time. By replacing the standard collapsable with a more subtle "Show logs" button, this UI can be cleaned up. In this case it also makes sense to always show this button even in non-advanced mode.

### PR DESCRIPTION
**Pull request by flash1293**
Fixes #20238 

## What

Improves the look of the logs in the connector form.

Old:
Screenshot 2022-12-08 at 17 19 15

New:
Screenshot 2023-01-02 at 17 14 47
Screenshot 2023-01-02 at 17 14 52

## How

* Remove the `JobItem` wrapper and use `JobLogs` directly - as there is only ever a single entry it doesn't make sense to wrap it into an accordion which looks like a list entry
* Use a button to toggle logs visibility (not tied to advanced mode anymore)

## Drive-by fix

In the old implementation there was a weird edge case:
* Go to an existing connector configuration
* Change the config so it will fail and hit "Save changes and test"
* Tests are running and showing an error as expected
* Click "Retest source" which tests the saved state of the connector
* Test succeeds, but the log is still showing the result of the failed connection test with the local changes

This PR is clearing out the log when the connector is re-tested

Also fixes some inconsistent spacing between components. The whole connector form controls should be cleaned up, but that will be a separate PR.

### CODE REVIEW

**Review by lmossman — APPROVED** [review]
Had a few comments but nothing majorly blocking. Lmk if you want me to take another look if you decide to replace the button.

Tested locally and it works as expected

File: airbyte-webapp/src/views/Connector/ConnectorCard/ConnectorCard.tsx [hunk 1]
```diff
@@ -219,7 +229,19 @@ export const ConnectorCard: React.FC<ConnectorCardCreateProps | ConnectorCardEdi
             />
           )}
           {/* Show the job log only if advanced mode is turned on or the actual job failed (not the check inside the job) */}
-          {job && (advancedMode || !job.succeeded) && <JobItem job={job} />}
+          {job && (
+            <div className={styles.connectionTestLogs}>
+              <Button
```
[the remark this answers is not in the document]
Comment (flash1293) [reply airbyte-webapp/src/views/Connector/ConnectorCard/ConnectorCard.tsx:234]: Agreed, changed the button to look like this:
Screenshot 2023-01-06 at 13 32 25
Screenshot 2023-01-06 at 13 32 28
